### PR TITLE
array-reduce - Simon Kim.

### PR DESCRIPTION
**Pull request by Skim-329**
Screen Shot 2021-10-15 at 1 15 01 PM

### CODE REVIEW

File: array-reduce/main.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = (previousValue, currentValue) => previousValue + currentValue;
+console.log(numbers.reduce(sum));
+
+const product = (previousValue, currentValue) => previousValue * currentValue;
+console.log(numbers.reduce(product));
```
Comment (TimelikeClosure) [inline array-reduce/main.js:24]: `sum` and `product` should be the final sum and product (respectively), not the callback functions.

---

image

File: array-reduce/main.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  .reduce((acc, amt) => acc + amt));
+console.log(balance);
+
+const composite =
+(traits.map(traits => traits.color)) +
+  (traits.map(traits => traits.type)) +
+  (traits.map(traits => traits.name)) +
+  (traits.map(traits => traits.level)) +
+  (traits.map(traits => traits.trainer));
+console.log(composite);
```
Comment (TimelikeClosure) [inline array-reduce/main.js:41]: Please try each of these using exactly one `reduce` each, with no `filter`s or `map`s:

---

image

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by Skim-329** [comment]
Screen Shot 2021-10-16 at 3 00 27 PM

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍

**Pull request merged**